File: rb/features.rb
# EmailValidation SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module EmailValidationFeatures
  def self.make_feature(name)
    case name
    when "base"
      EmailValidationBaseFeature.new
    when "ratelimit"
      EmailValidationRatelimitFeature.new
    when "retry"
      EmailValidationRetryFeature.new
    when "test"
      EmailValidationTestFeature.new
    when "timeout"
      EmailValidationTimeoutFeature.new
    else
      EmailValidationBaseFeature.new
    end
  end
end

File: py/emailvalidation_sdk/features.py
# EmailValidation SDK feature factory

from emailvalidation_sdk.feature.base_feature import EmailValidationBaseFeature
from emailvalidation_sdk.feature.test_feature import EmailValidationTestFeature


def _make_feature(name):
    features = {
        "base": lambda: EmailValidationBaseFeature(),
        "test": lambda: EmailValidationTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
